R/build-home-authors.R

```diff
@@ -105,22 +105,49 @@ author_list <- function(x, authors_info, comment = FALSE) {
   roles <- paste0(role_lookup[x$role], collapse = ", ")
   substr(roles, 1, 1) <- toupper(substr(roles, 1, 1))
 
+  orcid <- x$comment %>% purrr::pluck('ORCID')
+  x <- remove_orcid_comment(x)
+
   list(
     name = name,
     roles = roles,
-    comment = x$comment
+    comment = x$comment,
+    orcid = orcid_link(orcid)
   )
 }
 
 author_desc <- function(x, comment = TRUE) {
   paste(
     x$name,
     "<br />\n<small class = 'roles'>", x$roles, "</small>",
+    if (!is.null(x$orcid))
+      x$orcid,
     if (comment && !is.null(x$comment))
       paste0("<br/>\n<small>(", x$comment, ")</small>")
   )
 }
 
+remove_orcid_comment <- function(x) {
+
+  if (length(x$comment) == 1 && 'ORCID' %in% names(x$comment)) {
```

Comment:
Why doesn't `x$comment[["ORCID"]] <- NULL` work?

Reply:
I think because `x$comment` is a character vector when it exists (and explicit `NULL` when it does not).